### PR TITLE
Adds Toolchain mermaid chart

### PR DESCRIPTION
**Pull request by TheAngryByrd**
Adds an overview of the toolchain for Ionide/FSAC on index page.

image

### CODE REVIEW

File: generators/partials/footer.fsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ let footer =
         script [Src "/static/js/modernizr.custom-3.6.0.js"] []
         script [Src "/static/js/learn.js"] []
         script [Src "/static/js/hugo-learn.js"] []
-        link [Rel "stylesheet"; Href "/static/mermaid/mermaid.css"]
-        script [Src "/static/mermaid/mermaid.js"] []
+        // link [Rel "stylesheet"; Href "/static/mermaid/mermaid.css"]
+        script [Src "https://cdnjs.cloudflare.com/ajax/libs/mermaid/9.3.0/mermaid.min.js"] []
```
Comment (TheAngryByrd) [inline generators/partials/footer.fsx:25]: Mermaid was out of date, used this cdn, but I can download it locally if desired.

File: static/js/hugo-learn.js [hunk 2]
```diff
@@ -22,11 +22,13 @@ var images = $("div#body-inner img").not(".inline");
 images.wrap(function(){
   var image =$(this);
   var o = getUrlParameter(image[0].src);
-  var f = o['featherlight'];
-  // IF featherlight is false, do not use feather light
-  if (f != 'false') {
-    if (!image.parent("a").length) {
-      return "<a href='" + image[0].src + "' data-featherlight='image'></a>";
+  if(o) {
```
Comment (TheAngryByrd) [inline static/js/hugo-learn.js:25]: o could be undefined and was causing issues.

File: static/js/hugo-learn.js [hunk 3]
```diff
@@ -89,8 +91,8 @@ $(document).ready(function(){
 //       e.clearSelection();
 //       $(e.trigger).attr('aria-label', 'Link copied to clipboard!').addClass('tooltipped tooltipped-s');
 //   });
-//   $('code.language-mermaid').each(function(index, element) {
-//     var content = $(element).html().replace(/&amp;/g, '&');
-//     $(element).parent().replaceWith('<div class="mermaid" align="center">' + content + '</div>');
-//   });
-// });
+  $('code.language-mermaid').each(function(index, element) {
```
Comment (TheAngryByrd) [inline static/js/hugo-learn.js:94]: had to add this back in to get mermaid to transform from pre tag.

File: content/index.md [hunk 4]
```diff
@@ -23,3 +23,32 @@ Ionide is an organization which builds **high quality, cross platform developer
...
+
+    FsAutocomplete ||--|| Ionide-ProjInfo : "To get information for the F# Compiler"
+    FsAutocomplete ||--|| Ionide-LanguageServerProtocol  : "To communicate with an LSPClient"
+    FsAutocomplete ||--|| Ionide-FSharpSDKAnalyzers : "To run analyzers"
+    FsAutocomplete ||--|| "FSharp.Compiler.Service" : "To get type information, errors, warnings, etc."
+    FsAutocomplete ||--|| Fantomas : "To format code."
+    FsAutocomplete ||--|| FsharpLint : "To lint code."
+    
+    "ionide-vscode-fsharp" ||--||  FsAutocomplete : "Uses LSP to communicate to"
+    "ionide-vscode-fsharp" ||--||  Ionide-FsGrammer : "To colorize"
```
Comment (baronfel) [inline content/index.md]:
```suggestion
    "ionide-vscode-fsharp" ||--||  Ionide-FsGrammar : "To colorize"
```

Comment (baronfel) [reply content/index.md]: I also wonder if it's interesting to note that tools like highlight.js use Ionide-FsGrammar's textmate grammar to work.

**Review by baronfel — APPROVED** [review]

File: content/index.md (hunk 4 above)
Comment (TheAngryByrd) [reply content/index.md]: I don't know much about it so any more docs on that repo would be great!

**Comment by Krzysztof-Cieslak** [comment]
Hmmm...

image

**Pull request merged**